File: 191-number-of-1-bits/number-of-1-bits.cpp
class Solution {
public:
    int hammingWeight(int n) {
        // counting the set bit only 
        int count =0 ;
        while (n){
            if (n&1){
                count ++ ;
            }
            n = n>> 1;
        }
        return count ; 
    }
};

File: 191-number-of-1-bits/number-of-1-bits.js
/**
 * @param {number} n
 * @return {number}
 */
var hammingWeight = function(n) {
    // brute force approach 
    var count =0 ; 
    while (n){
        if (n&1)count ++ ; 
         n = n >> 1;
    }
    return count ; 
};